File: src/main/java/mods/cybercat/gigeresque/common/entity/ai/brain/memory/MemoryModuleTypes.java
package mods.cybercat.gigeresque.common.entity.ai.brain.memory;

import java.util.List;

import mods.cybercat.gigeresque.common.entity.impl.AlienEggEntity;
import mods.cybercat.gigeresque.common.entity.impl.FacehuggerEntity;
import mods.cybercat.gigeresque.mixins.common.entity.ai.MemoryModuleTypeInvoker;

import net.minecraft.entity.LivingEntity;
import net.minecraft.entity.ai.brain.MemoryModuleType;
import net.minecraft.util.math.BlockPos;

public class MemoryModuleTypes {
	private MemoryModuleTypes() {
	}

	public static final MemoryModuleType<LivingEntity> EGGMORPH_TARGET = MemoryModuleTypeInvoker
			.register("eggmorph_target");
	public static final MemoryModuleType<BlockPos> NEAREST_ALIEN_WEBBING = MemoryModuleTypeInvoker
			.register("nearest_alien_webbing");
	public static final MemoryModuleType<List<AlienEggEntity>> NEAREST_EGGS = MemoryModuleTypeInvoker
			.register("nearest_eggs");
	public static final MemoryModuleType<List<FacehuggerEntity>> NEAREST_FACEHUGGERS = MemoryModuleTypeInvoker
			.register("nearest_facehuggers");
	public static final MemoryModuleType<List<LivingEntity>> NEAREST_HOSTS = MemoryModuleTypeInvoker
			.register("nearest_hosts");
	public static final MemoryModuleType<BlockPos> NEAREST_LIGHT_SOURCE = MemoryModuleTypeInvoker
			.register("nearest_light_source");
}

File: src/main/kotlin/com/bvanseg/gigeresque/common/entity/ai/brain/memory/MemoryModuleTypes.kt
package com.bvanseg.gigeresque.common.entity.ai.brain.memory

import com.bvanseg.gigeresque.common.entity.impl.AlienEggEntity
import com.bvanseg.gigeresque.common.entity.impl.FacehuggerEntity
import com.bvanseg.gigeresque.mixins.MemoryModuleTypeInvoker
import net.minecraft.entity.LivingEntity
import net.minecraft.entity.ai.brain.MemoryModuleType
import net.minecraft.util.math.BlockPos

/**
 * @author Boston Vanseghi
 */
object MemoryModuleTypes {
    val EGGMORPH_TARGET: MemoryModuleType<LivingEntity> = MemoryModuleTypeInvoker.register("eggmorph_target")
    val NEAREST_ALIEN_WEBBING: MemoryModuleType<BlockPos> = MemoryModuleTypeInvoker.register("nearest_alien_webbing")
    val NEAREST_EGGS: MemoryModuleType<List<AlienEggEntity>> = MemoryModuleTypeInvoker.register("nearest_eggs")
    val NEAREST_FACEHUGGERS: MemoryModuleType<List<FacehuggerEntity>> = MemoryModuleTypeInvoker.register("nearest_facehuggers")
    val NEAREST_HOSTS: MemoryModuleType<List<LivingEntity>> = MemoryModuleTypeInvoker.register("nearest_hosts")
    val NEAREST_LIGHT_SOURCE: MemoryModuleType<BlockPos> = MemoryModuleTypeInvoker.register("nearest_light_source")
}